addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/DefaultRequestHandler.java

```diff
@@ -0,0 +1,221 @@
+                break;
+        }
+    }
+
+    @Override
+    public void checkUsage() {
+        WebRequester status = new WebRequester(endpointUsage, config);
+        logger.debug("Checking {} usage", thing.getUID().toString());
+        WebRequesterChecker result = status.process();
+        switch (result.status()) {
+            case WebRequester.REQ_STATUS_OK:
+                // Get Ink Levels
+                NodeList consumableInk = result.document().getDocumentElement()
+                        .getElementsByTagName("pudyn:Consumable");
+
+                try {
+                    for (int i = 0; i < consumableInk.getLength(); i++) {
+                        Element currInk = (Element) consumableInk.item(i);
+
+                        String inkName = currInk.getElementsByTagName("dd:MarkerColor").item(0).getTextContent();
+
+                        String consumeType = currInk.getElementsByTagName("dd:ConsumableTypeEnum").item(0)
+                                .getTextContent();
+
+                        if (consumeType.equalsIgnoreCase("printhead")) {
+                            continue;
+                        }
+
+                        String inkRemaining = currInk.getElementsByTagName("dd:ConsumableRawPercentageLevelRemaining")
+                                .item(0).getTextContent();
+
+                        logger.debug("{} Ink {} remaining level {}", thing.getUID().toString(), inkName, inkRemaining);
+
+                        switch (inkName.toLowerCase()) {
+                            case "black":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_BLACK_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                            case "yellow":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_YELLOW_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                            case "magenta":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_MAGENTA_LEVEL),
+                                        new DecimalType(inkRemaining));
+
+                                break;
+
+                            case "cyan":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_CYAN_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                            case "cyanmagentayellow":
+                                updateState(new ChannelUID(thing.getUID(), CGROUP_INK, CHANNEL_COLOUR_LEVEL),
+                                        new DecimalType(inkRemaining));
+                                break;
+
+                        }
+                    }
+                } catch (Exception ex) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Also you should always log the exception at least at some level, so you don't have to recompile the binding to trace a certain bug/error. Just pass it as the final argument to the logging statement.